qa-tester/assets/src/js/admin-bar/components/container/index.js

```diff
@@ -0,0 +1,305 @@
+			buildOptions: this.state.buildOptions.concat( branchOptions ),
+		} );
+	}
+
+	/**
+	 * Append to the default build options a list of PRs that have build zips available for download.
+	 */
+	async addPullRequestOptions() {
+		const pullRequests = await getPullRequestsWithBuilds();
+
+		// Create build options from list of PRs with plugin builds.
+		const prOptions = pullRequests.map( ( pr ) => {
+			return {
+				label: `PR #${ pr.number }: ${ pr.title }`,
+				value: pr.number,
+				origin: 'pr',
+			};
+		} );
+
+		this.setState( {
+			buildOptions: this.state.buildOptions.concat( prOptions ),
+		} );
+	}
+
+	/**
+	 * Set the selected build option.
+	 *
+	 * @param {Object} newOption Selected build option.
+	 */
+	handleChangeBuildOption( newOption ) {
+		this.setState( { buildOption: newOption } );
+	}
+
+	/**
+	 * Set whether a dev build is needed or not.
+	 *
+	 * @param {Event<HTMLInputElement>} event Change event.
+	 */
+	handleChangeDevBuild( event ) {
+		const checked = event.target.checked;
+		this.setState( { isDevBuild: checked } );
+	}
+
+	/**
+	 * Installs the selected build option.
+	 */
+	handleInstallation() {
+		const { isDevBuild, buildOption } = this.state;
+		this.setState( { isInstalling: true } );
+
+		const data = {
+			id: buildOption.value,
+			origin: buildOption.origin,
+			isDev: isDevBuild,
+		};
+
+		if ( buildOption.url ) {
+			data.url = buildOption.url;
+		}
+
+		apiFetch( {
+			path: '/amp-qa-tester/v1/install',
+			method: 'POST',
+			data,
+		} )
+			.then( () => {
+				window.location.reload();
+			} )
+			.catch( ( error ) => {
+				if ( error.message ) {
+					this.showError( error.message );
+				} else {
+					this.showError();
+				}
+			} );
+	}
+
+	/**
+	 * Display an error message.
+	 *
+	 * @param {string} message Message.
+	 */
+	showError( message = '' ) {
+		if ( ! message ) {
+			message = __(
+				'An unknown error occurred while installing build',
+				'amp-qa-tester'
+			);
+		}
+
+		this.setState( {
+			isInstalling: false,
+			error: message,
+		} );
+	}
+
+	render() {
+		const {
+			isLoading,
+			isInstalling,
+			error,
+			buildOption,
+			buildOptions,
+		} = this.state;
+
+		if ( isLoading ) {
+			return (
+				<>
+					<div className="loading" />
+					<p>{ __( 'Fetching build options…', 'amp-qa-tester' ) }</p>
+				</>
+			);
+		}
+
+		return (
+			<>
+				<BuildSelector
+					buildOptions={ buildOptions }
+					onOptionSelect={ this.handleChangeBuildOption }
+				/>
+
+				{ 'release' !== buildOption.origin && (
+					<div className="checkbox">
+						<input
+							id="is-development-build"
+							type="checkbox"
+							checked={ this.state.isDevBuild }
+							onChange={ this.handleChangeDevBuild }
+						/>
+
+						<label htmlFor="is-development-build">
+							{ __( 'Development build', 'amp-qa-tester' ) }
+						</label>
+					</div>
+				) }
+
+				{ /*TODO: Prevent submenu from closing after clicking button to activate selection and hovering away from it */ }
```

Comment:
Yes, this would be a nice to have. The submenu should remain expanded while installing.

Reply:
Done in https://github.com/ampproject/amp-wp/pull/4770/commits/ff08ea2465d51afe8304ab20c7ec7ed1ed6d2cae.